api/src/main/java/org/eclipse/microprofile/jwt/algorithm/KeyEncryptionAlgorithm.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2016-2017 Contributors to the Eclipse Foundation
+ *
+ *  See the NOTICE file(s) distributed with this work for additional
+ *  information regarding copyright ownership.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  You may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.jwt.algorithm;
+
+/**
+ * JWT JSON Web Key Encryption (Management) Algorithms.
+ * 
+ * @see <a href="https://tools.ietf.org/html/rfc7518#section-4">https://tools.ietf.org/html/rfc7518#section-4</a>
+ */
+public enum KeyEncryptionAlgorithm {
+    RSA_OAEP("RSA-OAEP"),
+    RSA_OAEP_256("RSA-OAEP-256"),
+    ECDH_ES("ECDH-ES"),
+    ECDH_ES_A256KW("ECDH-ES+A256KW"),
+    A256KW("A256KW");
```

Comment:
And missing (in the open world scenario)

A128KW, A192KW, dir, ECDH-ES+A128KW, ECDH-ES+A192KW, A128GCMKW, A192GCMKW, A256GCMKW, PBES2-HS256+A128KW, PBES2-HS384+A192K, PBES2-HS512+A256KW. (and possibly others)

Reply:
@rdebusscher I'm sorry, this comment is not productive. There is an effort there to limit a set of algorithms to only those which are `recommended` by the JWA spec. New ones can be added upon the demand if needed

Reply:
@rdebusscher I'll be open to trimming this list but these algorithm enums will not put any burden on the implementations 100%. The goal is type the minimum list of `required` algorithms from the JWA spec, so that users don't type the algorithm names manually, and, at the MP JWT level, we'll only require to support the very few ones.
Signature: `RS256` (easy:-) ), `ES256` - you won't even have to change your impl code to support it and it is added to support MP JWT servers accepting such tokens. `HS256` - the users are signing the JWT cookies but also would be good to verify them with MP JWT. It is about extending MP JWT reach.